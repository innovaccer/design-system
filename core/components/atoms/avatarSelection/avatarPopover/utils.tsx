import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
