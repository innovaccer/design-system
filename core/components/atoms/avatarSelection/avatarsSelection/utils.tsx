import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
