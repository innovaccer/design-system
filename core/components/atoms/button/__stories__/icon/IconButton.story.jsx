import { Button } from '@/index';

export const iconButton = () => (
  <Button appearance="basic" icon="keyboard_arrow_right" aria-label="Next in rank" tooltip="Next in rank" />
);

export default {
  title: 'Components/Button/Button/Icon/Icon Button',
  component: Button,
  parameters: {
    docs: {
      docPage: {
        title: 'Button',
        description: 'Secondary button with icon.',
        a11yProps: `**aria-label:** Add \`aria-label='Next in rank'\` to describe the action of button. `,
      },
    },
  },
};
