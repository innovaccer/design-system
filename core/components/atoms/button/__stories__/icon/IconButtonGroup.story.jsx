import Button from '@/components/atoms/button';

export const iconButtonGroup = () => (
  <div className="d-inline-flex">
    <Button className="mr-2" size="tiny" icon="content_copy" aria-label="Copy" tooltip="Copy" />
    <Button className="mr-2" size="tiny" icon="content_paste" aria-label="Paste" tooltip="Paste" />
    <Button size="tiny" icon="delete" aria-label="Delete" tooltip="Delete" />
  </div>
);

export default {
  title: 'Components/Button/Button/Icon/Icon Button Group',
  component: Button,
  parameters: {
    docs: {
      docPage: {
        title: 'Button',
        description: 'A pattern using tiny icon buttons in a group.',
        a11yProps: `**aria-label:** <br/> - Add \`aria-label='Copy'\` on button with *copy* icon to describe the action of button.<br/>- Add \`aria-label='Paste'\` on button with *paste* icon to describe the action of button.<br/>- Add \`aria-label='Delete'\` on button with *delete* icon to describe the action of button.`,
      },
    },
  },
};
