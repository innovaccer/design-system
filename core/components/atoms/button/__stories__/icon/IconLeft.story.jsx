import { Button } from '@/index';

export const iconLeft = () => (
  <Button icon="get_app" iconAlign="left" aria-label="Download">
    Download
  </Button>
);

export default {
  title: 'Components/Button/Button/Icon/Icon Left',
  component: Button,
  parameters: {
    docs: {
      docPage: {
        title: 'Button',
        description: 'Secondary button with left aligned icon.',
        a11yProps: ` **aria-label:** Add \`aria-label='Download'\` on button to indicate its purpose. `,
      },
    },
  },
};
