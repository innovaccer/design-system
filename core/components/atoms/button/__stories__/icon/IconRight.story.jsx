import { Button } from '@/index';

export const iconRight = () => (
  <Button icon="arrow_forward" iconAlign="right" aria-label="Next in rank">
    Next in rank
  </Button>
);

export default {
  title: 'Components/Button/Button/Icon/Icon Right',
  component: Button,
  parameters: {
    docs: {
      docPage: {
        title: 'Button',
        description: 'Secondary button with right aligned icon.',
        a11yProps: ` **aria-label:** Add \`aria-label='Next in rank'\` to describe the action of button. `,
      },
    },
  },
};
