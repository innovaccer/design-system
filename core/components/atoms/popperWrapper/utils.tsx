// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
