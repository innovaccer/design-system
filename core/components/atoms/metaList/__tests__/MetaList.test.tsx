import * as React from 'react';
import { render } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import { MetaList } from '@/index';
import { MetaListProps as Props } from '@/index.type';
import { Appearance as IconAppearance } from '../../icon';
import { Appearance as TextAppearance } from '../../text';

const appearance: IconAppearance[] = [
  'primary',
  'alert',
  'warning',
  'success',
  'accent1',
  'accent2',
  'accent3',
  'accent4',
  'default',
  'disabled',
  'info',
  'subtle',
  'white',
  'destructive',
  'primary_lighter',
  'primary_dark',
  'alert_lighter',
  'alert_dark',
  'warning_lighter',
  'warning_dark',
  'success_lighter',
  'success_dark',
  'accent1_lighter',
  'accent1_dark',
  'accent2_lighter',
  'accent2_dark',
  'accent3_lighter',
  'accent3_dark',
  'accent4_lighter',
  'accent4_dark',
  'inverse'
];

const labelAppearance: TextAppearance[] = [
  'success',
  'default',
  'disabled',
  'link',
  'subtle',
  'white',
  'destructive'
];

const booleanValues = [true, false];

const listWithTwoElements = [
  { icon: 'assessment', label: 'Meta data' },
  { icon: 'assessment', label: 'Meta data two' }
];

const getIconAppearance = (iconColor: string) => {
  const x = iconColor.indexOf('_');
  return iconColor.slice(0, x) + iconColor.charAt(x + 1).toUpperCase() + iconColor.slice(x + 2);
};

describe('MetaList component prop:seperatorAppearance options', () => {
  const mapper = {
    seperatorAppearance: valueHelper(appearance, { iterate: true, required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(
        <MetaList
          {...attr}
          list={[{ icon: 'assessment', label: 'Meta data' }, { icon: 'assessment', label: 'Meta data' }]}
        />

      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('MetaList component prop:IconAppearance options', () => {
  const mapper = {
    IconAppearance: valueHelper(appearance, { iterate: true, required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(
        <MetaList
          {...attr}
          list={[{ icon: 'assessment', label: 'Meta data' }]}
        />

      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('MetaList component prop:labelAppearance options', () => {
  const mapper = {
    labelAppearance: valueHelper(labelAppearance, { iterate: true, required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(
        <MetaList
          {...attr}
          list={[{ icon: 'assessment', label: 'Meta data' }, { icon: 'assessment', label: 'Meta data' }]}
        />

      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('MetaList component prop:seperator options', () => {
  const mapper = {
    seperator: valueHelper(booleanValues, { iterate: true, required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(
        <MetaList
          {...attr}
          list={[{ icon: 'assessment', label: 'Meta data' }]}
        />

      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('MetaList component with prop:list', () => {

  it('renders list when one element is given', () => {
    const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} seperator={true} />);
    expect(getByTestId('DesignSystem-MetaList--MetaLabel').textContent).toMatch('Meta data');
  });

  it('renders list when more than one elements are given', () => {
    const { getAllByTestId } = render(<MetaList list={listWithTwoElements} />);
    expect(getAllByTestId('DesignSystem-MetaList--MetaLabel')[0].textContent).toMatch('Meta data');
    expect(getAllByTestId('DesignSystem-MetaList--MetaLabel')[1].textContent).toMatch('Meta data two');
  });

  it('renders list when no element is given', () => {
    const { getByTestId } = render(<MetaList list={[]} />);
    expect(getByTestId('DesignSystem-MetaList')).not.toContainHTML('DesignSystem-MetaList--Meta');
  });

  it('renders list when label is not given', () => {
    const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: '' }]} />);
    expect(getByTestId('DesignSystem-MetaList--MetaLabel').textContent).toMatch('');
  });
});

describe('MetaList component with prop:Icon', () => {
  it('render the icon prop if present', () => {
    const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} />);
    expect(getByTestId('DesignSystem-MetaList--MetaIcon')).toBeInTheDocument();
  });

  it('check for icon prop when it is not present', () => {
    const { getByTestId } = render(<MetaList list={[{ label: 'Meta data' }]} />);
    expect(getByTestId('DesignSystem-MetaList')).not.toContainHTML('DesignSystem-MetaList--MetaIcon');
  });

  it('check for icon prop when list is empty', () => {
    const { getByTestId } = render(<MetaList list={[]} />);
    expect(getByTestId('DesignSystem-MetaList')).not.toContainHTML('DesignSystem-MetaList--MetaIcon');
  });

  it('check for icon prop when list contain more than one element', () => {
    const { getAllByTestId } = render(<MetaList list={[{ label: 'Meta data' }, { icon: 'assessment', label: 'Meta data two' }, { label: 'Meta data three' }]} />);
    expect(getAllByTestId('DesignSystem-MetaList--MetaIcon')).toHaveLength(1);
  });
});

describe('MetaList component with prop:separator', () => {
  it('check for separator prop when it is true', () => {
    const { getByTestId } = render(<MetaList list={listWithTwoElements} seperator={true} />);
    expect(getByTestId('DesignSystem-MetaList--Seperator')).toBeInTheDocument();
  });

  it('check for separator prop when it is false', () => {
    const { getByTestId } = render(<MetaList list={listWithTwoElements} seperator={false} />);
    expect(getByTestId('DesignSystem-MetaList')).not.toContainHTML('DesignSystem-MetaList--Seperator');
  });

  it('check for separator prop when it is enabled & list is empty', () => {
    const { getByTestId } = render(<MetaList list={[]} seperator={true} />);
    expect(getByTestId('DesignSystem-MetaList--Seperator')).toBeInTheDocument();
  });
});

describe('MetaList component with rightSeparator element', () => {

  it('check for rightSeparator prop when list is empty', () => {
    const { getByTestId } = render(<MetaList list={[]} />);
    expect(getByTestId('DesignSystem-MetaList')).not.toContainHTML('DesignSystem-MetaList--rightSeperator');
  });

  it('check for rightSeparator prop when list contain one element', () => {
    const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} />);
    expect(getByTestId('DesignSystem-MetaList')).not.toContainHTML('DesignSystem-MetaList--rightSeperator');
  });

  it('check for rightSeparator prop when list contain more than one element', () => {
    const { getAllByTestId } = render(<MetaList list={listWithTwoElements} seperator={true} />);
    expect(getAllByTestId('DesignSystem-MetaList--rightSeperator')).toHaveLength(1);
  });
});

describe('MetaList component tagName', () => {
  it('renders tagName for parent element', () => {
    const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} seperator={true} />);
    expect(getByTestId('DesignSystem-MetaList').tagName).toMatch('DIV');
  });

  it('renders tagName for child element', () => {
    const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} seperator={true} />);
    expect(getByTestId('DesignSystem-MetaList--Meta').tagName).toMatch('SPAN');
  });
});

describe('MetaList Component with overwrite class', () => {
  it('overwrite MetaList class', () => {
    const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} className="MetaListClass" />);
    expect(getByTestId('DesignSystem-MetaList')).toHaveClass('MetaListClass');
  });
});

describe('MetaList component with different prop:seperatorAppearance', () => {
  appearance.forEach((color) => {
    const colorOption = color && color.includes('_') ? getIconAppearance(color) : color;

    it(`should have the Icon--${color} class when seperatorAppearance=${color} `, () => {
      const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} seperator={true} seperatorAppearance={color}/>);
      expect(getByTestId('DesignSystem-MetaList--Seperator')).toHaveClass(`Icon--${colorOption}`);
    });
  });
});

describe('MetaList component with different prop:IconAppearance', () => {
  appearance.forEach((color) => {
    const colorOption = color && color.includes('_') ? getIconAppearance(color) : color;

    it(`should have the Icon--${color} class when IconAppearance=${color} `, () => {
      const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} iconAppearance={color}/>);
      expect(getByTestId('DesignSystem-MetaList--MetaIcon')).toHaveClass(`Icon--${colorOption}`);
    });
  });
});

describe('MetaList component with different prop:labelAppearance', () => {
  labelAppearance.forEach((color) => {
    it(`should have the Text--${color} class when labelAppearance=${color} `, () => {
      const { getByTestId } = render(<MetaList list={[{ icon: 'assessment', label: 'Meta data' }]} labelAppearance={color}/>);
      expect(getByTestId('DesignSystem-MetaList--MetaLabel')).toHaveClass(`Text--${color}`);
    });
  });
});
