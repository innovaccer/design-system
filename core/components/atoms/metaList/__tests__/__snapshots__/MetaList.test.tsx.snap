// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaList component prop:seperator options 
  seperator: false, size: "regular"
 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MetaList"
        data-test="DesignSystem-MetaList"
      >
        <span
          class="MetaList-item"
        >
          <span
            class="Meta"
            data-test="DesignSystem-MetaList--Meta"
          >
            <i
              class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
              data-test="DesignSystem-MetaList--MetaIcon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              assessment
            </i>
            <span
              class="Text Text--subtle Text--regular"
              data-test="DesignSystem-MetaList--MetaLabel"
            >
              Meta data
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MetaList"
      data-test="DesignSystem-MetaList"
    >
      <span
        class="MetaList-item"
      >
        <span
          class="Meta"
          data-test="DesignSystem-MetaList--Meta"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
            data-test="DesignSystem-MetaList--MetaIcon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            assessment
          </i>
          <span
            class="Text Text--subtle Text--regular"
            data-test="DesignSystem-MetaList--MetaLabel"
          >
            Meta data
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MetaList component prop:seperator options 
  seperator: false, size: "small"
 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MetaList"
        data-test="DesignSystem-MetaList"
      >
        <span
          class="MetaList-item"
        >
          <span
            class="Meta"
            data-test="DesignSystem-MetaList--Meta"
          >
            <i
              class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
              data-test="DesignSystem-MetaList--MetaIcon"
              role="button"
              style="font-size: 12px; width: 12px;"
            >
              assessment
            </i>
            <span
              class="Text Text--subtle Text--small"
              data-test="DesignSystem-MetaList--MetaLabel"
            >
              Meta data
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MetaList"
      data-test="DesignSystem-MetaList"
    >
      <span
        class="MetaList-item"
      >
        <span
          class="Meta"
          data-test="DesignSystem-MetaList--Meta"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
            data-test="DesignSystem-MetaList--MetaIcon"
            role="button"
            style="font-size: 12px; width: 12px;"
          >
            assessment
          </i>
          <span
            class="Text Text--subtle Text--small"
            data-test="DesignSystem-MetaList--MetaLabel"
          >
            Meta data
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MetaList component prop:seperator options 
  seperator: true, size: "regular"
 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MetaList"
        data-test="DesignSystem-MetaList"
      >
        <i
          class="material-symbols material-symbols-rounded Icon Icon--disabled MetaList-seperator MetaList-seperator--left"
          data-test="DesignSystem-MetaList--Seperator"
          role="button"
          style="font-size: 8px; width: 8px;"
        >
          fiber_manual_record
        </i>
        <span
          class="MetaList-item"
        >
          <span
            class="Meta"
            data-test="DesignSystem-MetaList--Meta"
          >
            <i
              class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
              data-test="DesignSystem-MetaList--MetaIcon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              assessment
            </i>
            <span
              class="Text Text--subtle Text--regular"
              data-test="DesignSystem-MetaList--MetaLabel"
            >
              Meta data
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MetaList"
      data-test="DesignSystem-MetaList"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--disabled MetaList-seperator MetaList-seperator--left"
        data-test="DesignSystem-MetaList--Seperator"
        role="button"
        style="font-size: 8px; width: 8px;"
      >
        fiber_manual_record
      </i>
      <span
        class="MetaList-item"
      >
        <span
          class="Meta"
          data-test="DesignSystem-MetaList--Meta"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
            data-test="DesignSystem-MetaList--MetaIcon"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            assessment
          </i>
          <span
            class="Text Text--subtle Text--regular"
            data-test="DesignSystem-MetaList--MetaLabel"
          >
            Meta data
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MetaList component prop:seperator options 
  seperator: true, size: "small"
 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MetaList"
        data-test="DesignSystem-MetaList"
      >
        <i
          class="material-symbols material-symbols-rounded Icon Icon--disabled MetaList-seperator MetaList-seperator--left"
          data-test="DesignSystem-MetaList--Seperator"
          role="button"
          style="font-size: 8px; width: 8px;"
        >
          fiber_manual_record
        </i>
        <span
          class="MetaList-item"
        >
          <span
            class="Meta"
            data-test="DesignSystem-MetaList--Meta"
          >
            <i
              class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
              data-test="DesignSystem-MetaList--MetaIcon"
              role="button"
              style="font-size: 12px; width: 12px;"
            >
              assessment
            </i>
            <span
              class="Text Text--subtle Text--small"
              data-test="DesignSystem-MetaList--MetaLabel"
            >
              Meta data
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MetaList"
      data-test="DesignSystem-MetaList"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--disabled MetaList-seperator MetaList-seperator--left"
        data-test="DesignSystem-MetaList--Seperator"
        role="button"
        style="font-size: 8px; width: 8px;"
      >
        fiber_manual_record
      </i>
      <span
        class="MetaList-item"
      >
        <span
          class="Meta"
          data-test="DesignSystem-MetaList--Meta"
        >
          <i
            class="material-symbols material-symbols-rounded Icon Icon--subtle Meta-icon"
            data-test="DesignSystem-MetaList--MetaIcon"
            role="button"
            style="font-size: 12px; width: 12px;"
          >
            assessment
          </i>
          <span
            class="Text Text--subtle Text--small"
            data-test="DesignSystem-MetaList--MetaLabel"
          >
            Meta data
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
